### PR TITLE
Support for secp256k1 algorithms (AKA ES256K)

### PR DESCRIPTION
**Pull request by captHarlock69**
### Changes

Added support for the ES256K algorithm, which is the ECDSA using the secp256k1 elliptic curve according to [jwt.io](https://jwt.io/) specifications.

### Testing

Added unit tests, they cover all tests present for ES256 curve, they all have an ES256K label.

### CODE REVIEW

File: lib/src/test/java/com/auth0/jwt/JWTCreatorTest.java [hunk 1]
```diff
@@ -157,7 +158,42 @@ public void shouldNotOverwriteKeyIdIfAddedFromRSAAlgorithms() throws Exception {
...
+        ECPrivateKey privateKey = (ECPrivateKey) PemUtils.readPrivateKeyFromFile(PRIVATE_KEY_FILE_EC_256K
+                , "EC");
+        ECDSAKeyProvider provider = mock(ECDSAKeyProvider.class);
+        when(provider.getPrivateKeyId()).thenReturn("my-key-id");
+        when(provider.getPrivateKey()).thenReturn(privateKey);
+        
+        String signed = JWTCreator.init()
+                .sign(Algorithm.ECDSA256K(provider));
+        
+        System.out.println("TOKEN: "+signed);
```
Comment (captHarlock69) [inline lib/src/test/java/com/auth0/jwt/JWTCreatorTest.java:173]:
```suggestion
        System.out.println("TOKEN: "+signed); 
```
LEFT BY MISTAKE, should be removed.

**Comment by lbalmaceda** [comment]
@captHarlock69 Thanks for this. Looking into how to reduce the introduced lines of code, is it me or beside the algorithm name, everything else looks just like the regular ES256 verification?

**Comment by captHarlock69** [comment]
> @captHarlock69 Thanks for this. Looking into how to reduce the introduced lines of code, is it me or beside the algorithm name, everything else looks just like the regular ES256 verification?

Yes that's correct, the code i basically the same beside key/certs files
I preferred adding a dedicated test in order to have a clear test interface.

**Comment by lbalmaceda** [comment]
Superseded by #439

**Pull request closed without merging**